components/file_system/src/io_stats/biosnoop.rs

```diff
@@ -146,22 +147,26 @@ pub fn get_io_type() -> IOType {
     unsafe { *IDX.with(|idx| IO_TYPE_ARRAY[idx.0]) }
 }
 
-pub(crate) fn fetch_io_bytes(mut io_type: IOType) -> IOBytes {
+pub fn fetch_io_bytes() -> [IOBytes; IOType::COUNT] {
+    let mut bytes = Default::default();
     unsafe {
         if let Some(ctx) = BPF_CONTEXT.as_mut() {
-            let io_type_buf_ptr = &mut io_type as *mut IOType as *mut u8;
-            let mut io_type_buf =
-                std::slice::from_raw_parts_mut(io_type_buf_ptr, std::mem::size_of::<IOType>());
-            if let Ok(e) = ctx.stats_table.get(&mut io_type_buf) {
-                assert!(e.len() == std::mem::size_of::<IOBytes>());
-                return std::ptr::read_unaligned(e.as_ptr() as *const IOBytes);
+            for io_type in IOType::iter() {
+                let io_type_buf_ptr = &mut io_type as *mut IOType as *mut u8;
+                let mut io_type_buf =
+                    std::slice::from_raw_parts_mut(io_type_buf_ptr, std::mem::size_of::<IOType>());
+                if let Ok(e) = ctx.stats_table.get(&mut io_type_buf) {
+                    assert!(e.len() == std::mem::size_of::<IOBytes>());
+                    bytes[io_type as usize] =
+                        std::ptr::read_unaligned(e.as_ptr() as *const IOBytes);
+                }
             }
         }
     }
-    IOBytes::default()
+    bytes
 }
 
-pub fn init_io_snooper() -> Result<(), String> {
+pub fn init() -> Result<(), String> {
```

Comment:
consider a more clear name?

Reply:
`init` belongs to `io_stats` crate. So I think it wouldn't be confusing.

Reply:
just a suggestion, usually it will be used `init()` not `biosnoop::init()`.

Reply:
biosnoop or proc or stub is all private implementation details. We only expose `io_stats` crate.